config/subsidies-consumer/subsidy-attachments-dispatching/file-processor.js

```diff
@@ -0,0 +1,203 @@
+ * 
+ * This function loops deltas and matches strings starting with:
+ *  - <data:// (subsidie meta files)
+ *  - <share:// (subsidie attachments)
+ *  
+ * When found, get a cookie from producer if needed, then either download or delete the file. 
+ * Download function contains retry mechanism.
+ *
+ * @param {Array} termObjects - An array of delta objects to be processed.
+ * @param {function} fetch - The fetch function used for network requests.
+ * @param {string} operation - The operation to perform:  DOWNLOAD_OPERATION or DELETE_OPERATION.
+ * @returns {Promise} A promise representing the completion of the processing.
+ */
+async function processFileDeltas(termObjects, fetch, operation) {
+  
+  for (const item of termObjects) {
+    // Process meta files (<data://)
+    if (isMetaFile(item)) {
+      if (operation === DOWNLOAD_OPERATION) {
+        downloadFile(item.object.replace('<data://', '<share://subsidies/'), fetch);
+      } else if (operation === DELETE_OPERATION) {
+        deleteFile(item.object.replace('<data://', '<share://subsidies/'));
+      }
+    }
+
+    // Process attachments (<share://)
+    else if (isAttachmentFile(item)) {
+      if (operation === DOWNLOAD_OPERATION) {
+        downloadFile(item.subject, fetch);
+      } else if (operation === DELETE_OPERATION) {
+        deleteFile(item.subject);
+      }
+    }
+  }
+}
+
+/**
+ * Download a file with specified URI from producer.
+ * @param {string} uri - The URI of the file to be downloaded.
+ * @param {function} fetcher - The fetch function to use for downloading.
+ * @returns {Promise} A promise representing the completion of the download.
+ */
+async function downloadFile(uri, fetcher) {
+  await loginIfNeeded();
+
+  const fetchOptions = {
+    headers: {
+      cookie: cookie,
+    },
+  };
+
+  uri = uri.replace(/[<>]/g, "");
+  const fileName = uri.replace('share://', '');
+  const downloadFileURL = `${SYNC_BASE_URL}/delta-files-share/download?uri=${uri}`;
+
+  let filePath = `/share/${fileName}`;
+
+  let attempt = 1;
+
+  while (attempt <= MAX_FILE_DOWNLOAD_RETRY_ATTEMPTS) {
+    console.log(`Downloading file ${uri} from ${downloadFileURL}, Attempt ${attempt}`);
+    const response = await fetcher(downloadFileURL, fetchOptions);
+
+    if (response.ok) {
+      const buffer = await response.buffer();
+      await createDirectories(filePath);
+
+      fs.writeFileSync(filePath, buffer);
+      return;
+    } else {
+      console.error(`Failed to download file ${uri} (${response.status})`);
+      await retryAfterDelay(SLEEP_TIME_AFTER_FAILED_FILE_DOWNLOAD_OPERATION);
+      attempt++;
+    }
+  }
+
+  console.error(`Exceeded maximum retry attempts for downloading file ${uri}`);
+}
+
+/**
+ * Delete a file with specified URI.
+ * @param {string} uri - The URI of the file to be deleted.
+ */
+function deleteFile(uri) {
+  uri = uri.replace(/[<>]/g, "");
+  const fileName = uri.replace('share://', '');
+  const filePath = `/share/${fileName}`;
```

Comment:
```suggestion
  const filePath = uri.replace('share://', '/share/');
```
we don't need filename

Reply:
Indeed we dont